### PR TITLE
introduce syscalls

### PR DESCRIPTION
**Pull request by rob-solana**
#### Problem
 programs need to consult bank state

 #### Summary of Changes
 introduce convention for exposing bank state to programs via a synthesized data account

part 1 of #4246

### CODE REVIEW

File: sdk/src/account_utils.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+//! useful extras for Account state
+use crate::account::{Account, KeyedAccount};
+use crate::instruction::InstructionError;
+use bincode::ErrorKind;
+
+/// Conveinence trait to covert bincode errors to instruction errors.
```
Comment (mvines) [inline sdk/src/account_utils.rs]: nit: Conveinence

File: sdk/src/syscall/slothashes.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
+//! named accounts for synthesized data accounts for bank state, etc.
+//!
+//! this account carries the Bank's most recent blockhashes for some N parents
+//!
+use crate::account::Account;
+use crate::hash::Hash;
+use crate::pubkey::Pubkey;
+
+/// "Sysca11SlotHashes11111111111111111111111111"
```
Comment (CriesofCarrots) [inline sdk/src/syscall/slothashes.rs]: @rob-solana , why the ones in "Sysca11" here and above, instead of `l`s? Just to be cute?

Comment (mvines) [reply sdk/src/syscall/slothashes.rs]: Base58 has no `l` :-/

Comment (CriesofCarrots) [reply sdk/src/syscall/slothashes.rs]: Oh, duh!

**Pull request merged**